### ISSUE TITLE
Nomad UI showing 'vUnknown'

### ISSUE THREAD

**Issue by bjornicus** [hashicorp/nomad#10672]
### Nomad version
Nomad v1.1.0 (2678c3604bc9530014208bc167415e167fd440fc)

### Operating system and Environment details
Linux 4.15.0-1002-azure-fips #2-Ubuntu SMP Wed May 27 01:47:10 UTC 2020 x86_64 x86_64 x86_64 GNU/Linux

### Issue
The nomad UI is showing the version as `vUknown`:
![image](https://user-images.githubusercontent.com/1729048/120095434-cecee080-c0da-11eb-9cec-340ce26d170e.png)

### Reproduction steps
unsure.

#### Expected Result
it should show V1.1.0
#### Actual Result
it shows vUnknown

---

And as I'm finishing writing this up I refreshed the UI and it now shows the expected v1.1.0.  What I think happened is I had upgraded the server nodes to 1.1.0 but not the clients yet.  I just finished adding new clients with 1.1.0 and removed all of the old (1.0.4) clients.  Does vUnknown just mean I have a mix of different versions?  Being connected to the server I was expecting to see the server nomad version in the UI.

**Comment by tgross** [comment]
Hi @bjornicus! I'm not one of our UI experts but it looks like this is bubbling up from [`app/services/system.js#L42`](https://github.com/hashicorp/nomad/blob/v1.1.0/ui/app/services/system.js#L42) which is a call to [`/v1/agent/self`](https://www.nomadproject.io/api-docs/agent#query-self). What was happening here probably depends on a bit on which agent you were getting the web UI from. Was that from one of the new servers?

**Comment by the-maldridge** [comment]
I have observed this on a homogeneous 1.1.0 fleet as well.  At first I thought it might be because my binaries did not have all of the version symbols defined at build time, but I have also observed this occasionally with Hashicorp provided binaries in my testing cluster which is also 1.1.0.

**Comment by bjornicus** [comment]
I'm not actualy sure if I was getting the UI from.  We have a load balancer in front of them so I'd get a random one of 3 servers.  I think I did connect directly to a specific instance (running new) as well to check, but I'm not certain.  I also think even after upgrading all 3 servers I was still seeing this until the clients also were upgraded, based on what I wrote above, which means I definitely would have been getting UI from the new server.

FWIW, I have definitely seen vUnknown previously, so this is not a new issue with 1.1.0.
